vortex-array/src/arrays/struct_/compute/zip.rs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::ops::{BitAnd, BitOr, Not};
+
+use vortex_error::VortexResult;
+use vortex_mask::Mask;
+
+use crate::arrays::{StructArray, StructVTable};
+use crate::compute::{ZipKernel, ZipKernelAdapter, zip};
+use crate::validity::Validity;
+use crate::vtable::ValidityHelper;
+use crate::{Array, ArrayRef, register_kernel};
+
+impl ZipKernel for StructVTable {
+    fn zip(
+        &self,
+        if_true: &StructArray,
+        if_false: &dyn Array,
+        mask: &Mask,
+    ) -> VortexResult<Option<ArrayRef>> {
+        let Some(if_false) = if_false.as_opt::<StructVTable>() else {
+            return Ok(None);
+        };
+
+        assert_eq!(if_true.names(), if_false.names()); // this is properly checked above
```

Comment:
```suggestion
        assert_eq!(if_true.names(), if_false.names()); // this is checked in the entry point function
```
Where is this checked?

Reply:
i think you're seeing the outdated version? 

<img width="427" height="166" alt="image" src="https://github.com/user-attachments/assets/8dd1b47f-631b-4792-b915-6ff1d0280011" />